src/tests/ftest/cart/utest/utest_portnumber.c

```diff
@@ -0,0 +1,286 @@
+/*
+ * (C) Copyright 2020 Intel Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+ * The Government's rights to use, modify, reproduce, release, perform, display,
+ * or disclose this software are subject to the terms of the Apache License as
+ * provided in Contract No. 8F-30005.
+ * Any reproduction of computer software, computer software documentation, or
+ * portions thereof marked with this legend must also reproduce the markings.
+ */
+/**
+ * This file is part of CaRT testing.
+ */
+
+/*
+ * This code tests whether a provider returns an error if 2 independent
+ * instances attempts to open the same port number.
+ *
+ * Two child process are forked with the same provider information.
+ * The test is setup so that first child opens the port and then
+ * the second child should fail.
+ * Note, synchronization between child process is performed via sleep
+ * function calls.  The sleep time should prevent any problems, but
+ * beware if an issue arise where the results are swapped.
+ */
+/*
+ * Reference jira ticket DAOS-5732 to include socket and verb tests.
+ */
+#define MY_TESTS_NOT_INCLUDED
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <stdarg.h>
+#include <stddef.h>
+#include <setjmp.h>
+#include <time.h>
+#include <sys/wait.h>
+
+#include <sys/types.h>
+#include <sys/shm.h>
+#include <sys/ipc.h>
+#include <semaphore.h>
+#include <string.h>
+#include <errno.h>
+
+#include <cmocka.h>
+#include <cart/api.h>
+#include "gurt/debug.h"
+
+/* global semaphores */
+sem_t *child1_sem;
+sem_t *child2_sem;
+
+int shmid_c1;
+int shmid_c2;
```

Comment:
minor, those probably should be static to not leak outside in symbols

Reply:
This is a program and not a library that someone will be linking with.  These symbols will not be leaked.